nsqd/nsqd.go

```diff
@@ -203,13 +206,14 @@ func (n *NSQD) getFlag(f int32) bool {
 }
 
 func (n *NSQD) SetHealth(err error) {
-	n.errMtx.Lock()
-	defer n.errMtx.Unlock()
-	n.err = err
 	if err != nil {
 		n.setFlag(flagHealthy, false)
```

Comment:
I don't think we need `flagHealthy` at all anymore - `IsHealthy` can become:

```go
func (n *NSQD) IsHealthy() bool {
    return n.GetError() == nil
}
```

Right?